frontend/src/components/common/ViewButton.tsx

```diff
@@ -0,0 +1,44 @@
+import eyeIcon from '@iconify/icons-mdi/eye';
+import eyeOff from '@iconify/icons-mdi/eye-off';
+import Icon from '@iconify/react';
+import { IconButton, Tooltip } from '@material-ui/core';
+import React from 'react';
+import { KubeObject } from '../../lib/k8s/cluster';
+import EditorDialog from './EditorDialog';
+
+interface ViewButtonProps {
+    item: KubeObject;
+}
+
+function ViewButton(props: ViewButtonProps) {
+  const {item} = props;
+  const [toggle, setToggle] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState<string>('');
+  function handleButtonClick() {
+    setToggle((toggle) => !toggle);
+  }
+  return (
+    <>
+      <Tooltip title="View YAML">
+        <IconButton
+          edge="end"
+          aria-label="toggle field visibility"
+          onClick={handleButtonClick}
+          onMouseDown={event => event.preventDefault()}
+        >
+          <Icon icon={toggle ? eyeOff : eyeIcon} />
```

Comment:
We don't really need a different icon because we have a modal anyway when the editor is opened. It's a small change but let's remove it.